var/spack/repos/builtin/packages/knem/package.py

```diff
@@ -11,11 +11,11 @@ class Knem(AutotoolsPackage):
     communication for large messages."""
 
     homepage = "http://knem.gforge.inria.fr"
-    url = "http://gforge.inria.fr/frs/download.php/37186/knem-1.1.3.tar.gz"
-    list_url = "http://knem.gforge.inria.fr/download"
+    url = "https://gitlab.inria.fr/knem/knem/uploads/4a43e3eb860cda2bbd5bf5c7c04a24b6/knem-1.1.4.tar.gz"
```

Comment:
This URL won't work for version 1.1.3 since Spack has no way of knowing the particular hash to use. If there is no option to download from a URL without a hash, we'll need a version-specific URL for each version. Also, I wouldn't get rid of `list_url`, it's used for things like `spack versions` and `spack checksum`.